### PR TITLE
AnonCreds Credentials using the W3C Standard - proof value encoding

### PR DESCRIPTION
**Pull request by Artemkaaas**
Use message pack for encoding proof values of w3c credentials and presentation.
Results of proof value string length on example of credentials signature:
* base64(JSON) previously used - 3821 
* base64(**MessagePack**) will be used - 2541
* base64(CBOR) - 3041

### CODE REVIEW

**Review by TimoGlastra — APPROVED** [review]

File: src/data_types/nonce.rs [hunk 1]
```diff
@@ -187,9 +196,22 @@ impl<'a> Deserialize<'a> for Nonce {
...
             }
+
+            fn visit_seq<E>(self, mut seq: E) -> Result<Self::Value, E::Error>
+            where
+                E: SeqAccess<'a>,
+            {
+                let mut vec = Vec::new();
+
+                while let Ok(Some(Value::Number(elem))) = seq.next_element() {
+                    vec.push(elem.as_u64().unwrap() as u8);
```
Comment (andrewwhitehead) [inline src/data_types/nonce.rs:207]: This should likely be more tolerant of bad inputs, not panicking via `unwrap`. It should also return an error if a non-Number element is found.

Comment (Artemkaaas) [reply src/data_types/nonce.rs]: fixed

**Review by andrewwhitehead — APPROVED** [review]

**Comment by andrewwhitehead** [comment]
Tests are failing because a Rust dependency now requires 1.66 or higher. We can either update the CI rust version or add a lockfile in another PR.

**Comment by swcurran** [comment]
Seems like updating would be the prudent thing?  What is typically done in this situation?  How much risk is bumping the rust version?  As I recall — the. #295  is failing one test (Android) because it needs 1.70.  Should we bump to that?

**Comment by andrewwhitehead** [comment]
There's no real risk in bumping the Rust version used, it just means that we can't guarantee compatibility with older versions. I think the Android failure is different (one of cross's dependencies), and partly fixed by this PR, but it would eventually need a lockfile to maintain its platform support.

**Comment by swcurran** [comment]
OK — just trying to figure out how we get to a decision/action on this. Who makes the call?

**Comment by TimoGlastra** [comment]
@berendsliedrecht just checking if this can impose a problem for the supported versions of the RN wrapper if the version is bumped?

**Comment by swcurran** [comment]
@Artemkaaas --- sorry to be a pain. Looks like we have one unit test failing. Could you please take a look?

> test data_types::w3c::proof::tests::encoded_object_encode_decode_works ... FAILED

Fails on both Ubuntu and Windows.

**Pull request merged**